imap_processing/swapi/swapi_utils.py

```diff
@@ -0,0 +1,76 @@
+import collections
+from enum import IntEnum
+
+import xarray as xr
+
+
+class SWAPIAPID(IntEnum):
+    """Create ENUM for apid.
+
+    Parameters
+    ----------
+    IntEnum : IntEnum.
+    """
+
+    SWP_HK = 1184
+    SWP_SCI = 1188
+    SWP_AUT = 1192
+
+
+def add_metadata_to_array(data_packet, metadata_arrays):
+    """Add metadata to the metadata_arrays.
+
+    Parameters
+    ----------
+    data_packet : space_packet_parser.parser.Packet
+        SWE data packet
+    metadata_arrays : dict
+        metadata arrays.
+    """
+    for key, value in data_packet.header.items():
+        metadata_arrays.setdefault(key, []).append(value.raw_value)
+    for key, value in data_packet.data.items():
+        metadata_arrays.setdefault(key, []).append(value.raw_value)
+
+
+def create_dataset(packets):
+    """Create dataset for each metadata field.
+
+    Parameters
+    ----------
+    packets : list
+        packet list
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all metadata field data in xr.DataArray.
+    """
+    metadata_arrays = collections.defaultdict(list)
+
+    for data_packet in packets:
+        add_metadata_to_array(data_packet, metadata_arrays)
+
+    epoch_time = xr.DataArray(
+        metadata_arrays["SHCOARSE"],
+        name="Epoch",
+        dims=["Epoch"],
+        attrs=dict(
+            description="Mission elapsed time",
+            units="seconds since start of the mission",
+        ),
+    )
+
+    dataset = xr.Dataset(
+        coords={"Epoch": epoch_time},
+    )
+
+    # create xarray dataset for each metadata field
+    for key, value in metadata_arrays.items():
+        if key == "SHCOARSE":
+            continue
+        dataset[key] = xr.DataArray(
+            value,
+            dims=["Epoch"],
+        )
```

Comment:
This is adding a DataArray into the single dataset you created before. I think this is typically done by passing it into the `data_vars` creation?
```suggestion
    # create xarray DataArray for each metadata field
    data_vars = {key: xr.DataArray(value, dims=["Epoch"]) for key, value in metadata_arrays.items() if key != "SHCOARSE"}
```

Reply:
I wanted to add those to the `dataset` and return dataset. If I make the suggested change, it creates dictionary of xr.Datarray. I would need to go through that again to add to the dataset, right?

Reply:
I meant to add this to the `xr.Dataset(data_vars=..., coords=...)` line above, but didn't comment on enough lines.

Reply:
so you were thinking like this
```
    data_vars = {key: xr.DataArray(value, dims=["Epoch"]) for key, value in metadata_arrays.items() if key != "SHCOARSE"}

    dataset = xr.Dataset(
        data_vars=data_vars,
        coords={"Epoch": epoch_time},
    )
```

Reply:
Exactly, or adding the data_vars dict comprehension in the Dataset creation if it is easier to read everything together rather than the intermediate variable. Either way looks good to me though!